### PR TITLE
[luci/pass] Introduce QuantizeWeightsPass

### PR DESCRIPTION
**Pull request by glistening**
It introduces QuantizeWeightsPass, which calls WeigghtsOnly visitor.

ONE-DCO-1.0-Signed-off-by: Sanggyu Lee <sg5.lee@samsung.com>

Related: https://github.com/Samsung/ONE/pull/11073, https://github.com/Samsung/ONE/issues/11057

### CODE REVIEW

**Review by hseok-oh — APPROVED** [review]
LGTM

File: compiler/luci/pass/src/QuantizeWeightsPass.test.cpp [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+
+TEST(QuantizeWeightsPassTest, name)
+{
+  luci::QuantizeWeightsPass pass(loco::DataType::FLOAT32, loco::DataType::U8,
+                                 luci::QuantizationGranularity::LayerWise);
+  auto const name = pass.name();
+  ASSERT_NE(nullptr, name);
+}
+
+TEST(QuantizeWeightsPassTest, name_ctx)
```
Comment (seanshpark) [inline compiler/luci/pass/src/QuantizeWeightsPass.test.cpp:29]: plz add number of negative tests same or more than positive tests

**Comment by glistening** [comment]
@seanshpark I added 2 negative tests.

@hseok-oh PTAL

**Review by jinevening — DISMISSED** [review]
LGTM with minor comment

**Review by jinevening — APPROVED** [review]
LGTM

**Review by seanshpark — APPROVED** [review]

File: compiler/luci/pass/src/QuantizeWeightsPass.test.cpp [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+  luci::QuantizeWeightsPass pass(std::move(ctx));
+  auto const name = pass.name();
+  ASSERT_NE(nullptr, name);
+}
+
+TEST_F(QuantizeWeightsPassTest, run_u8_NEG)
+{
+  loco::Graph g;
+  luci::QuantizeWeightsPass pass(loco::DataType::FLOAT32, loco::DataType::U8,
+                                 luci::QuantizationGranularity::ChannelWise);
```
Comment (hseok-oh) [inline compiler/luci/pass/src/QuantizeWeightsPass.test.cpp:110]: I think it's better to check input model type, and we may support U8 output model type someday...
But anyway, it's OK.

**Review by hseok-oh — APPROVED** [review]
LGTM

**Pull request merged**